Update User Guide to include Unavailability in worker-add and Reassign command (#116)
AY2021S1-CS2103-F10-4/tp: docs/UserGuide.md

Adds a new worker to the McScheduler.

> Format: `worker-add n/NAME hp/PHONE_NUMBER a/ADDRESS p/HOURLY_PAY [r/ROLE]... [u/UNAVAILABILITY]...`

Reviewer: ```suggestion
Format: `worker-add n/NAME hp/PHONE_NUMBER a/ADDRESS p/HOURLY_PAY [r/ROLE]... [u/UNAVAILABLE_DAY UNAVAILABLE_TIME]...`
```
Would this be clearer since all the params will be visible directly instead of having to refer to the details below? It is also more consistent with the format of specifying a role requirement and mass assigning.

Author: Yes, agreed! Will make the change

Author: Yes, agreed! Will make the change

* Adds a worker with the specified `NAME`, `PHONE_NUMBER`, `ADDRESS`, `HOURLY_PAY`, `ROLE`(s) and
`UNAVAILABILITY`(s).
* `HOURLY_PAY` can have a maximum of 2 decimal places only. 
* The worker will be fit to take on the specified `ROLE`(s) in a shift. The specified `ROLE`(s) must be an existing role
  in the McScheduler. A role can be added to the McScheduler using the [role-add](#adding-a-role-role-add) command.
* `UNAVAILABILITY` must be in the format {u/DAY TIME}. `DAY` should take one of these values: **Mon, Tue, Wed, Thu,
Fri, Sat, Sun**. These values are case-insensitive (i.e. `Mon`, `MON`, `mon`, `mOn` etc. are all accepted). `TIME` should

> take one of these values: **AM, PM, FULL**. These values are also case-insensitive (i.e. `aM`, `fUll`, `pm`, etc. are all

Reviewer: Assuming the suggestion above is taken, this can be split into 2 points for `UNAVAILABLE_DAY` and `UNAVAILABLE_TIME` which may be easier to read.

> accepted). The worker will be unavailable during the specified `UNAVAILABILITY`(s). A worker cannot be assigned to a shift
> they are unavailable for with the [assign](#assigning-a-worker-to-a-role-in-a-shift-assign) command.

Reviewer: ```suggestion
accepted). The worker will be unavailable during the specified `UNAVAILABILITY`(s). 
```
Perhaps the removed lines are more appropriate in the `assign` feature itself.

* Adds a shift on the specified `DAY` at the specified `TIME`.
* The day specified should take one of these values: **Mon, Tue, Wed, Thu, Fri, Sat, Sun**. These values are case-

>   insensitive i.e. `Mon`, `MON`, `mon` etc. are all accepted.

Reviewer: May I know the rationale behind this change since this example is used above for `UNAVAILABILITY` as well?

Author: I was supposed to delete it for `UNAVAILABILITY` as well but I missed it out; will fix it! Because our days are now enum values so we can't have numbers like 0 inside

Reviewer: I believe that's a capital 'O'

* Unassigns the worker at the specified `WORKER_INDEX` from the shift at the specified `SHIFT_ INDEX`. The indexes refer
  to the index numbers shown in the displayed worker and shift lists. The indexes **must be positive integers** i.e. 1,
  2, 3, …​
  
### Reassigning an existing assignment: `reassign`

Reassigns an existing assignment in the McScheduler, similar to an assignment edit.

Format: `reassign so/OLD_SHIFT_INDEX wo/OLD_WORKER_INDEX sn/NEW_SHIFT_INDEX wn/NEW_WORKER_INDEX r/ROLE`

* Reassigns an existing assignment using the indexes provided such that the worker at `NEW_WORKER_INDEX` will be assigned to the
shift at `NEW_SHIFT_INDEX` for the specified `ROLE`. The indexes **must be positive integers** i.e. 1,2, 3, …​

> * The old assignment involving the worker at the specified `OLD_WORKER_INDEX` and the shift at `OLD_SHIFT_INDEX` must be valid.

Reviewer: ```suggestion
* The old assignment involving the worker at the specified `OLD_WORKER_INDEX` and the shift at `OLD_SHIFT_INDEX` must exist.
```
Not sure if this will be clearer

The old assignment will be edited during a successful `reassign` call.
* The `OLD_WORKER_INDEX` can be the same as the `NEW_WORKER_INDEX`. The `OLD_SHIFT_INDEX` can be the same as the `NEW_SHIFT_INDEX`.
This allows workers to be reassigned to the same shift but under a different `ROLE`.
* No reassignment will be made if there already exists a duplicate assignment in the McScheduler.

> Example:

Reviewer: ```suggestion
Examples:
```

* `reassign so/4 wo/1 sn/4 wn/2 r/Chef` Reassigns the 2nd worker on the worker list to the 4th shift on the shift list as a Chef.

> * `reassign so/1 wo/2 sn/3 wn/2 r/Cashier` Reassigns the 2nd worker on the worker list to the 3rd shift on the shift list as a Cashier.

Reviewer: ```suggestion
* `reassign so/4 wo/1 sn/4 wn/2 r/Chef` Reassigns the 2nd worker on the worker list to the 4th shift on the shift list as a chef.

* `reassign so/1 wo/2 sn/3 wn/2 r/Cashier` Reassigns the 2nd worker on the worker list to the 3rd shift on the shift list as a cashier.
```
For consistency with the rest of the features

Author: All right, made the format change for some of the other commands as well!

Role | **List** | `role-list`
Assignment | **Assign** | `assign s/SHIFT_INDEX w/WORKER_INDEX r/ROLE`<br>e.g. `assign s/3 w/2 r/Cashier`
Assignment | **Unassign** | `unassign s/SHIFT_INDEX w/WORKER_INDEX`<br>e.g. `unassign s/4 w/1`

> Assignment | **Reassign** | `reassign so/OLD_SHIFT_INDEX wo/OLD_WORKER_INDEX sn/NEW_SHIFT_INDEX wn/NEW_WORKER_INDEX`<br>e.g. `reassign so/4 wo/1 sn/1 wo/1 r/Chef`

Reviewer: ```suggestion
Assignment | **Reassign** | `reassign so/OLD_SHIFT_INDEX wo/OLD_WORKER_INDEX sn/NEW_SHIFT_INDEX wn/NEW_WORKER_INDEX`<br>e.g. `reassign so/4 wo/1 sn/1 wn/1 r/Chef`
```

Author: Good catch!
